_source/_posts/2020-04-20-android-authentication.adoc

```diff
@@ -0,0 +1,508 @@
+---
+layout: blog_post
+title: "Build an Android Application with Authentication"
+author: daniel-pereira
+by: contractor
+communities: [mobile]
+description: "Tutorial: Build a simple Android application with authentication then add social and biometric authentication."
+tags: [oauth, oidc, kotlin, android, mobile, authentication]
+tweets:
+- "Learn how to add OAuth 2.0 authentication, biometric, and Social Auth to your Android applications📱"
+- "Need to add authentication to your Android app? Check out this post! 🔒"
+- "This post walks you through creating a new Android app with OAuth 2.0 auth💥"
+image: blog/featured/okta-android-skew.jpg
+type: conversion
+---
+:toc: macro
+:page-liquid:
+:experimental:
+
+With mobile apps becoming ever-present in users' lives, following best security practices has become essential in protecting your users and your apps. Implementing security alone and from the ground up can be costly for your development team, create a less-than-ideal user experience, and be susceptible to design/implementation errors. The easiest path to strong security, and a positive user experience, is to turn to the experts.
+
+The Okta OIDC SDK can help with this in many ways:
+- Our security experts take the development effort of creating a secure authentication method off your plate
+- OAuth 2.0 and OpenID Connect allows your users to easily authenticate on your app with social login, using industry-standard secure protocols
+
+This post will show you a simple example of how you can use Okta OIDC SDK to create a (Kotlin) Android app with authentication.
+
+toc::[]
+
+== (TLDR) Download the finished app with Android Authentication
+
+If you want to follow along using a completed version of the app,clone this repository from  https://github.com/oktadeveloper/okta-oidc-android-example[GitHub] using the `simplified` branch:
+
+[source,sh]
+----
+git clone https://github.com/oktadeveloper/okta-oidc-android-example.git -b simplified
+cd okta-oidc-android-example
+----
+
+NOTE: The finished app in the `master` branch from the repository includes everything described in this blog, along with  some additional UI and functionality, testing Okta OIDC easier, and providing you with a production-level skeleton to work with when creating your own app.
+
+== Creating your first Android app
```

Comment:
http://www.conissaunce.com/pages/infoq-title-case.html says this should be `Creating Your First Android App`. If I channel Lindsay, it should never have `ing`, so should be `Create Your First Android App`.